pkg/resources/external_table.go

```diff
@@ -144,204 +140,137 @@ func ExternalTable() *schema.Resource {
 	}
 }
 
-type externalTableID struct {
-	DatabaseName      string
-	SchemaName        string
-	ExternalTableName string
-}
-
-// String() takes in a externalTableID object and returns a pipe-delimited string:
-// DatabaseName|SchemaName|ExternalTableName.
-func (si *externalTableID) String() (string, error) {
-	var buf bytes.Buffer
-	csvWriter := csv.NewWriter(&buf)
-	csvWriter.Comma = externalTableIDDelimiter
-	dataIdentifiers := [][]string{{si.DatabaseName, si.SchemaName, si.ExternalTableName}}
-	if err := csvWriter.WriteAll(dataIdentifiers); err != nil {
-		return "", err
-	}
-	strExternalTableID := strings.TrimSpace(buf.String())
-	return strExternalTableID, nil
-}
-
-// externalTableIDFromString() takes in a pipe-delimited string: DatabaseName|SchemaName|ExternalTableName
-// and returns a externalTableID object.
-func externalTableIDFromString(stringID string) (*externalTableID, error) {
-	reader := csv.NewReader(strings.NewReader(stringID))
-	reader.Comma = externalTableIDDelimiter
-	lines, err := reader.ReadAll()
-	if err != nil {
-		return nil, fmt.Errorf("not CSV compatible")
-	}
-
-	if len(lines) != 1 {
-		return nil, fmt.Errorf("1 line at a time")
-	}
-	if len(lines[0]) != 3 {
-		return nil, fmt.Errorf("3 fields allowed")
-	}
-
-	externalTableResult := &externalTableID{
-		DatabaseName:      lines[0][0],
-		SchemaName:        lines[0][1],
-		ExternalTableName: lines[0][2],
-	}
-	return externalTableResult, nil
-}
-
 // CreateExternalTable implements schema.CreateFunc.
-func CreateExternalTable(d *schema.ResourceData, meta interface{}) error {
+func CreateExternalTable(d *schema.ResourceData, meta any) error {
```

Comment:
I've upgraded it, but I think It won't make sense If I add another one, because the existing one needs environment variables we cannot provide right now. I would leave it as it is and when we'll have needed resources we can add more tests with external sources.

Reply:
Okay, I did not consider that. You are right.